### PR TITLE
School Library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Bhekisile**
## Changes

- Created class Person with instance variables, constructors, getters, setters, and private and public methods.
- Created class Student that inherits from Person's class, having parent's constructor extended by classroom and a method called `play-hooky`.
- Created class Teacher that inherits from Person's class, having parent's constructor extended by specialization and overriding `can_use_services?`.
- Added `README.md` file

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    @id = generate_id
```
Comment (TedLivist) [inline person.rb]: - [ ] Only `name` and `parent_permission` parameters should be optional parameters. So `age` should be required.

- [ ] You have not implemented the `generate_id` method. I cannot find this method within this class. Kindly implement it or you can use `rand(1...1000)` to generate a random number

  ```
  def initialize(age, name = 'Unknown', parent_permission: true)
    @id = rand(1..1000)
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    super(name, age, parent_permission)
+    @classroom = classroom
+  end
```
Comment (TedLivist) [inline student.rb]: - [ ] The `classroom` should be passed as a parameter also.

- [ ] The `age` should not be optional

- [ ] The arrangement of arguments on line 7 should match the Person class.

- [ ] `parent_permission` should be well assigned in the super arguments.

  ```
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    super(name, age, parent_permission)
+    @specialization = specialization
+  end
```
Comment (TedLivist) [inline teacher.rb]: - [ ] The `specialization` should be passed as a parameter also.

- [ ] The `age` should not be optional

- [ ] The arrangement of arguments on line 7 should match the Person class.

- [ ] `parent_permission` should be well assigned in the super arguments.

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization= specialization
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Bhekisile,

Good job so far!
✔️ Classes are in different files based on convention
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @Bhekisile  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights
- Created individual files for `Person`, `Student`, and `Teacher`✔✔
- All class methods implemented correctly  :heavy_check_mark: 
- Correct git flow :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**